src/com/massivecraft/factions/engine/EnginePermBuild.java

```diff
@@ -209,6 +211,30 @@ public void buildEntity(EntityDamageByEntityEvent event)
 		build(player, entity.getLocation().getBlock(), event);
 	}
 	
+	@EventHandler(priority = EventPriority.NORMAL, ignoreCancelled = true)
+	public void hitEntity(ProjectileHitEvent event)
+	{
+		// If a player's projectile ...
+		ProjectileSource projectileSource = event.getEntity().getShooter();
+		if (!(projectileSource instanceof Entity) || MUtil.isntPlayer(projectileSource)) return;
+		MPlayer player = MPlayer.get(projectileSource);
+		
+		// ... is burning ...
+		Entity entityProjectile = event.getEntity();
+		if (entityProjectile == null || entityProjectile.getFireTicks() < 0) return;
+		
+		// ... and hits an entity which is edited on damage (and thus likely to burn) ...
+		Entity entityTarget = event.getHitEntity();
+		if (entityTarget == null || !EnumerationUtil.isEntityTypeEditOnDamage(entityTarget.getType())) return;
+		
+		// ... and the player can't build there
+		Block block = entityProjectile.getLocation().getBlock();
+		if (!isProtected(ProtectCase.BUILD, false, MPlayer.get(player), PS.valueOf(block), block)) return;
+		
+		// ... then remove the fire.
+		entityProjectile.setFireTicks(-1);
```

Comment:
Is extinguishing the projectile the way to go about this? My understanding is that if you were to fire an arrow at a normal block, the expected the arrow would be allowed to remain on fire itself. Is it possible to prevent the fire damage without changing this expected behavior?

Reply:
I've since reworked the approach to make use of EntityCombustByEntityEvent rather than ProjectileHitEvent.
Arrow's behavior is no longer modified, and entities such as the Armor Stand do not combust.